core-web/libs/edit-content/src/lib/fields/dot-edit-content-host-folder-field/store/host-folder-field.store.ts

```diff
@@ -0,0 +1,120 @@
+};
+
+const initialState: HostFolderFiledState = {
+    nodeSelected: null,
+    nodeExpaned: null,
+    tree: [],
+    status: 'idle'
+};
+
+export const HostFolderFiledStore = signalStore(
+    withState(initialState),
+    withComputed(({ status }) => ({
+        iconClasses: computed(() => {
+            const currentStatus = status();
+
+            return {
+                'pi-spin pi-spinner': currentStatus === 'pending',
+                'pi-chevron-down': currentStatus !== 'pending'
+            };
+        })
+    })),
+    withMethods((store, dotEditContentService = inject(DotEditContentService)) => ({
+        loadSites: rxMethod<string | null>(
+            pipe(
+                tap(() => patchState(store, { status: 'pending' })),
+                switchMap((path) => {
+                    return dotEditContentService
+                        .getSitesTreePath({ perPage: 7000, filter: '*' })
+                        .pipe(
+                            tapResponse({
+                                next: (sites) => patchState(store, { tree: sites }),
+                                error: () => patchState(store, { status: { error: 'fail' } }),
+                                finalize: () => patchState(store, { status: 'fulfilled' })
+                            }),
+                            map((sites) => ({
+                                path,
+                                sites
+                            }))
+                        );
+                }),
+                filter(({ path }) => !!path),
+                switchMap(({ path, sites }) => {
+                    const hasPaths = path.includes('/');
+                    if (!hasPaths) {
+                        const response: CustomTreeNode = {
+                            node: sites.find((item) => item.key === path),
+                            tree: null
+                        };
+
+                        return of(response);
+                    }
+
+                    return dotEditContentService.buildTreeByPaths(path);
+                }),
+                tap(({ node, tree }) => {
+                    const changes: Partial<HostFolderFiledState> = {};
+                    if (node) {
+                        changes.nodeSelected = node;
+                    }
+
+                    if (tree) {
+                        const currentTree = store.tree();
+                        const newTree = currentTree.map((item) => {
+                            if (item.key === tree.path) {
+                                return {
+                                    ...item,
+                                    children: [...tree.folders]
+                                };
+                            }
+
+                            return item;
+                        });
+                        changes.tree = newTree;
+                    }
+
+                    patchState(store, changes);
+                })
+            )
+        ),
```

Comment:
Can we granulate this in different functions? Looks really hard to read and maintain

Reply:
@zJaaal Yes on the same page with you I think is a lot of functions in one file, but the `ngrx/signals` works only with closure functions with `store` in all context, check this example.

![Screenshot 2024-06-14 at 12 53 33 PM](https://github.com/dotCMS/core/assets/7611944/36f7aaa8-4b94-481b-aef7-61941c3bbd66)

Here is an example to turn the store into a class, but the methods with rxMethod cannot be private methods instead should be treated like attrs.

![Screenshot 2024-06-14 at 12 57 32 PM](https://github.com/dotCMS/core/assets/7611944/341de69e-656e-468c-86ab-ce7dbb03c4da)

I need to do more research on this and create a standard with the frontend team.